### DISCUSSION TITLE
LibreChat, Azure and LangSmith

### DISCUSSION THREAD

**Discussion by aclem2payx** [danny-avila/LibreChat#3055] [Q&A]
TLDR:  Is anyone using LangSmith to capture traces from LibreChat running in an Azure Container App?

Details:

- LangSmith 0.5
    - Self hosted in Azure
    - Accepting traces from other LLM apps

- LibreChat 7.2
    - Azure Container App
    - ENV Vars set in Azure not in .env
        - LANGCHAIN_TRACING_V2=true
        - LANGCHAIN_ENDPOINT=https//our.self.hosted.langsmith/api
        - LANGCHAIN_HUB_API_URL=https//our.self.hosted.langsmith/api-hub
        - LANGCHAIN_API_KEY=an-appropriate-guid
        - LANGCHAIN_CALLBACKS_BACKGROUND=true
    - Various additional ENV vars also set in Azure, including
        - DEBUG_CONSOLE=true
        - MONGO_URI
        - OPENID_*

Behavior:

LangSmith is accepting traces from other LLM apps.
I can find no evidence of traces from LibreChat in the LangSmith logs.
LibreChat works great in all other respects, authorization, mongo integration and Azure APIM integration.
I can successfully curl LangSmith from the LibreChat container and I see the activity in LangSmith logs.
If I leave off the env var LANGCHAIN_CALLBACKS_BACKGROUND from the LibreChat container, I get a warning "Log: [WARN]: You have enabled LangSmith tracing without backgrounding callbacks."  I take this as a good sign that the env vars are being 'seen', however no traces.

Ideas and thoughts from the community?

**Comment by jakobwowy** [comment]
I also don't get any tracing with the Azure OpenAI API as an endpoint.
